packages/plugin-dts/src/dts.ts

```diff
@@ -133,6 +142,33 @@ export async function generateDts(data: DtsGenOptions): Promise<void> {
     ? distPath
     : rawCompilerOptions.declarationDir || './dist';
 
+  if (build) {
+    // do not allow to use bundle DTS when 'build: true' since temp declarationDir should be set by user in tsconfig
+    if (bundle) {
+      throw Error(`Can not set "dts.bundle: true" when "dts.build = true"`);
+    }
+
+    // can not set '--declarationDir' or '--outDir' when 'build: true'.
+    if (
+      (!rawCompilerOptions.outDir ||
+        normalize(rawCompilerOptions.outDir) !==
+          normalize(resolve(dirname(configPath), outDir))) &&
+      (!rawCompilerOptions.declarationDir ||
+        normalize(rawCompilerOptions.declarationDir) !==
+          normalize(resolve(dirname(configPath), outDir)))
+    ) {
+      const info =
+        rawCompilerOptions.outDir && !rawCompilerOptions.declarationDir
+          ? 'outDir'
+          : 'declarationDir';
+      throw Error(
+        `Please set ${info}: "${outDir}" in ${color.underline(
+          configPath,
+        )} to keep it same as Lib config.`,
```

Comment:
Indicate which config to align with.

Reply:
$outdir is the config value that should be configured, users just need to copy it from the terminal output.

Reply:
Yeah, I mean we should let user to know the detailed field.

Reply:
updated